bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -93,7 +96,7 @@ private Event createCommandEvent(String topic, String payload, String source) {
         ItemEventPayloadBean bean = deserializePayload(payload, ItemEventPayloadBean.class);
         Command command = null;
         try {
-            command = (Command) parse(bean.getType(), bean.getValue());
+            command = (Command) parse(bean.getType() + TYPE_POSTFIX, bean.getValue());
```

Comment:
My idea was to have `parse` as my internal API which parses class names to instances. I didnt want the method to know about the fact that we pass around truncated class names. To make that even more clear I changed the parameter to `simpleClassName`. This respects the open/closed principle in a way that we are open for Commands and States with different class name patterns and the method is closed for modification because it will always do what it says it does.
I see your point though, that we handle all class names the same in this case and parse may also be the one knowing about the truncation. wdyt?